mpd/mpd.go

```diff
@@ -485,6 +492,16 @@ func (as *AdaptationSet) addRepresentation(r *Representation) error {
 	return nil
 }
 
+func (as *AdaptationSet) AddNewRole(schemeIDURI string, value string) (*Role, error) {
+	r := &Role{
+		SchemeIDURI: Strptr(schemeIDURI),
+		Value:       Strptr(value),
+	}
+	r.AdaptationSet = as
+	as.Roles = append(as.Roles, r)
+	return r, nil
+}
```

Comment:
I'm not sure about returning nil errors all the time, but we do it all the time above.........